Entity/UserRole.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\Entity;
+
+use DateTime;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Security\Core\Role\Role;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table
+ * @ORM\HasLifecycleCallbacks
+ */
+class UserRole extends Role
+{
+    /**
+     * @var int
+     *
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
+     */
+    private $roleName;
+
+    /**
+     * @var DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    private $createdOn;
+
+    /**
+     * @var DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    private $editedOn;
+
+    /**
+     * @param string $roleName
+     */
+    public function __construct(string $roleName)
+    {
+        parent::__construct($roleName);
+
+        $this->roleName = $roleName;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getRole(): string
+    {
+        return $this->roleName;
+    }
+
+    /**
+     * @ORM\PrePersist
+     */
+    public function onPrePersist(): void
+    {
+        $this->createdOn = new DateTime();
+        $this->editedOn = new DateTime();
+    }
+
+    /**
+     * @ORM\PreUpdate
+     */
+    public function onPreUpdate(): void
+    {
+        $this->editedOn = new DateTime();
+    }
+
+    public function getCreatedOn(): DateTime
```

Comment:
You should group getters whenever possible imo, this should move to the top where the other getters are :)